### PR TITLE
Refactor tests to eliminate cooked

### PR DESCRIPTION
**Pull request by badpandabear**
Remove/change unit tests that require "cooked", plus a utility function that the dev-suite code uses.

### CODE REVIEW

File: pypeit/par/util.py [hunk 1]
```diff
@@ -718,3 +718,37 @@ def make_pypeit_file(pypeit_file, spectrograph, data_files, cfg_lines=None, setu
     msgs.info('PypeIt file written to: {0}'.format(pypeit_file))
 
 
+def make_tool_config(filename, cfg_lines, data_lines, block_type):
```
Comment (profxj) [inline pypeit/par/util.py:721]: Heads up that this method will disappear in
a PR on refactoring PypeIt file generation.

**Review by profxj — APPROVED** [review]
Thanks for pushing this through to the finish!

**Comment by profxj** [comment]
Let's not pull this in until the sister PR in the DevSuite is ready to merge as well.

**Comment by profxj** [comment]
Also, the test we are failling

_____________________________ test_wmko_flux_std ______________________________

may need to go to the DevSuite as it may need the Telluric file (too large for GitHub's CI)

**Comment by kbwestfall** [comment]
> Also, the test we are failling
> 
> _____________________________ test_wmko_flux_std ______________________________
> 
> may need to go to the DevSuite as it may need the Telluric file (too large for GitHub's CI)

I agree this should likely get moved to the devsuite, but it's worth noting this only failed for the windows tests.  And it's unclear to me what exactly causes the error.  Is there a basic test that checks if the caching system works with windows paths?

File: pypeit/tests/test_coremeta.py [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
 
 from pypeit.core import meta
 
+# TODO -- We could use *many* more tests on core/meta.py
+
 def test_meta():
+    # Constructr the simple dicg
     d = meta.get_meta_data_model()
     assert 'mjd' in d.keys(), 'MJD not in keys'
+    assert isinstance(d, dict)
```
Comment (kbwestfall) [inline pypeit/tests/test_coremeta.py:14]: Maybe put this before the previous assert?  I.e., check that `d` is a `dict` before executing `d.keys()`?

File: pypeit/tests/test_detector.py [hunk 3]
```diff
@@ -36,6 +35,7 @@
 
 def test_init():
     detector = detector_container.DetectorContainer(**def_det)
+    assert detector.specaxis == 1
```
Comment (kbwestfall) [inline pypeit/tests/test_detector.py:38]: Nice.

File: pypeit/tests/test_flatfield.py [hunk 4]
```diff
@@ -1,55 +1,23 @@
...
-from pypeit.tests.tstutils import dev_suite_required, load_kast_blue_masters, cooked_required
 from pypeit import flatfield
-from pypeit import slittrace
-from pypeit.spectrographs.util import load_spectrograph
-from pypeit.images import pypeitimage
 from pypeit import bspline
 
 def data_path(filename):
     data_dir = os.path.join(os.path.dirname(__file__), 'files')
     return os.path.join(data_dir, filename)
```
Comment (kbwestfall) [inline pypeit/tests/test_flatfield.py:18]: Thanks for cleaning up the repeat definitions of `data_path`!  Here's one you missed.

File: pypeit/tests/test_pydl.py [hunk 5]
```diff
@@ -2,11 +2,11 @@
 Module to run tests on pyidl functions
 """
 
-import numpy as np
 from pypeit.core import pydl
 import pytest
 
-# TODO: Add tests!
+# TODO: JFH -- any suggestions??
+
```
Comment (kbwestfall) [inline pypeit/tests/test_pydl.py:9]: Personally, I would remove this file.

File: pypeit/tests/test_telluric.py [hunk 6]
```diff
@@ -8,6 +8,8 @@
 from pypeit.tests.tstutils import telluric_required, tell_test_grid
 from pypeit import data
 
+#TODO -- Consider moving this into the DevSuite tesets
+
```
Comment (kbwestfall) [inline pypeit/tests/test_telluric.py]: I think anything that requires the telluric files should be moved the dev-suite.

**Review by kbwestfall — APPROVED** [review]
Excellent!  Thanks for going to the lengths of cleaning up the test file imports, as well.  Only a couple minor suggestions from me.  Approving now.

File: pypeit/tests/test_telluric.py (hunk 6 above)
Comment (badpandabear) [reply pypeit/tests/test_telluric.py]: Agreed, I moved this to the dev-suite

**Comment by badpandabear** [comment]
It looked to me like `test_wmko_flux_std` had succeeded downloading the telluric file, but failed due to being out of memory later on. I did test that it successfully ran on my local windows installation. I also moved it to the dev-suite unit_tests section.

File: pypeit/par/util.py (hunk 1 above)
Comment (badpandabear) [reply pypeit/par/util.py:721]: I was expecting that, and think the newer PypeIt file class is better anyway. There'll also be code in that dev-suite that needs to change.

**Pull request merged**